File: src/main/java/com/mock_json/api/models/RequestLog.java
package com.mock_json.api.models;

import jakarta.persistence.*;
import lombok.*;

import com.fasterxml.jackson.annotation.JsonBackReference;

import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "request_logs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RequestLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String url;

    @Column(nullable = false)
    private String method;

    @Column(nullable = false)
    private String ip;

    @Column(nullable = false)
    private int status;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "project_id")
    @JsonBackReference
    private Project project;

}

File: internal/models/request_log.go
package models

import (
	"time"
)

// RequestLog represents a request log in the system.
type RequestLog struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	CreatedAt time.Time `json:"created_at"`
	URL       string    `gorm:"column:url;not null" json:"url"` // Matched json tag with existing annotation
	Method    string    `gorm:"not null" json:"method"`
	IP        string    `gorm:"column:ip_address" json:"ip_address"` // Matched json tag with existing annotation
	Status    int       `gorm:"not null" json:"status"`
	ProjectID uint      `gorm:"not null" json:"project_id"`
	Project   *Project  `gorm:"foreignKey:ProjectID" json:"project,omitempty"`
}

// TableName specifies the table name for the RequestLog model.
func (RequestLog) TableName() string {
	return "request_logs"
}
